src/platform/Linux/ConnectivityManagerImpl.cpp

```diff
@@ -69,12 +75,17 @@ ConnectivityManagerImpl ConnectivityManagerImpl::sInstance;
 char ConnectivityManagerImpl::sWiFiIfName[];
 #endif
 
+WiFiDriver::ScanCallback * ConnectivityManagerImpl::mpScanCallback;
+NetworkCommissioning::Internal::WirelessDriver::ConnectCallback * ConnectivityManagerImpl::mpConnectCallback;
+
 CHIP_ERROR ConnectivityManagerImpl::_Init()
 {
 #if CHIP_DEVICE_CONFIG_ENABLE_WPA
     mWiFiStationMode              = kWiFiStationMode_Disabled;
     mWiFiStationReconnectInterval = System::Clock::Milliseconds32(CHIP_DEVICE_CONFIG_WIFI_STATION_RECONNECT_INTERVAL);
 #endif
+    mpConnectCallback = nullptr;
+    mpScanCallback    = nullptr;
```

Comment:
The management of the callback contexts here is a bit dangerous. I don't see anything that makes the API fail with some busy if the callbacks are already set.